File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    answer = 0
    graph = [[] for _ in range(n)]
    visited = [False] * (n)
    for i in range(len(computers)):
        for j in range(len(computers[0])):
            if computers[i][j] == 1 and i != j:
                graph[i].append(j)
    for i in range(n):
        if visited[i] == False:
            bfs(i,visited,graph)
            answer +=1
    return answer

def bfs(n,visited,graph):
    q = deque()
    q.append(n)
    visited[n] = True
    while q:
        x = q.popleft()
        for i in graph[x]:
            if not visited[i]:
                q.append(i)
                visited[i] = True

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    ArrayList<Integer>[] graph;
    boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        graph = new ArrayList[n];
        visited = new boolean[n];

        for (int i=0;i<n;i++) {
            graph[i] = new ArrayList<>();
        }
        for (int i=0;i<n;i++) {
            for (int j=0;j<n;j++) {
                if (computers[i][j] == 1 && i != j) {
                    graph[i].add(j);
                }
            }
        }
        
        for (int i=0; i<n; i++) {
            if (visited[i] != true) {
                answer +=1;
                bfs(i);
            }
        }
        return answer;
    }
    

public void bfs(int node) {
    Queue<Integer> q = new LinkedList<>();
    q.add(node);
    visited[node] = true;
    int now = 0;
    while (!q.isEmpty()) {
        now = q.poll();
        for (Integer i: graph[now]) {
            if (visited[i] != true) {
                q.add(i);
                visited[i] = true;
            }
        }
    }
}
}